File: netgear/RAX5/usr/lib/lua/webGetFunc/trafficMeter.lua
local M = { }

local uci = require "luci.model.uci".cursor()
local uci_st = require "luci.model.uci".cursor(nil, "/var/state")
local json = require "luci.json"
local sys  = require "luci.sys"
local netUtils = require "commonFunc.netUtils"
local log = require "luci.log" -- for debug


function M.getTrafficMeterEnable()
	local ret = uci:get("tm", "tm_control", "traffic_on")
        return ret
end

function M.getTmType()
	local volume_contrl = uci:get("tm", "tm_control", "volcontrl")
	local time_control = uci:get("tm", "tm_control", "timecontrl")

	if (volume_contrl == "true" and time_control == "false") then
	  return "volControl"
	elseif (volume_contrl == "false" and time_control == "true") then
	  return "timeControl"
	end
end

function M.getControlType()
	local ret = uci:get("tm", "tm_control", "contrl_type")
        return ret
end

function M.getVolMonthlyLimit()
	local ret = uci:get("tm", "tm_control", "volume_monthly_limit")
        return ret
end

function M.getVolRoundUp()
	local ret = uci:get("tm", "tm_control", "round_up_volume")
        return ret
end

function M.getTimeControlLimit()
	local ret = uci:get("tm", "tm_control", "conntime_monthly_limit")
        return ret
end

function M.getCounterDay()
	local ret = uci:get("tm", "tm_control", "day")
        return ret
end

function M.getCounterHour()
	local ret = uci:get("tm", "tm_control", "hour")
        return ret
end

function M.getCounterMin()
	local ret = uci:get("tm", "tm_control", "min")
        return ret
end

function M.getAmPmSel()
	local ret = uci:get("tm", "tm_control", "ampm_sel")
        return ret
end

function M.getPopMessage()
	local ret = uci:get("tm", "tm_control", "waterMark")
        return ret
end

function M.getTurnLed()
	local ret = uci:get("tm", "tm_control", "led_on")
        return ret
end

function M.getDisableInternet()
	local ret = uci:get("tm", "tm_control", "block_on")
        return ret
end

function M.getStartTime()
	local ret = uci_st:get("tm", "tm_statistics", "start_data_time")
	return ret
end

function M.getCurrentTime()
	local ret = luci.sys.exec("date")
	return ret
end

function M.getLeftTraffic()
	local ret = uci_st:get("tm", "tm_statistics", "volume_left")
	return ret
end

-- Today tm_statistics Start--
function M.getTodayConnectTime()
	local ret = uci_st:get("tm", "tm_statistics", "today_conntime")
	return ret
end

function M.getTodayUpload()
	local ret = uci_st:get("tm", "tm_statistics", "today_upload")
	return ret
end

function M.getTodayDownload()
	local ret = uci_st:get("tm", "tm_statistics", "today_download")
	return ret
end

function M.getTodayTotal()
	local ret = uci_st:get("tm", "tm_statistics", "today_total")
	return ret
end
-- Today tm_statistics End--

-- Yesterday tm_statistics Start--
function M.getYesterdayConnectTime()
	local ret = uci_st:get("tm", "tm_statistics", "yesterday_conntime")
	return ret
end

function M.getYesterdayUpload()
	local ret = uci_st:get("tm", "tm_statistics", "yesterday_upload")
	return ret
end

function M.getYesterdayDownload()
	local ret = uci_st:get("tm", "tm_statistics", "yesterday_download")
	return ret
end

function M.getYesterdayTotal()
	local ret = uci_st:get("tm", "tm_statistics", "yesterday_total")
	return ret
end
-- Yesterday tm_statistics End--

-- Thisweek tm_statistics Start--
function M.getThisweekConnectTime()
	local ret = uci_st:get("tm", "tm_statistics", "thisweek_conntime")
	return ret
end

function M.getThisweekUpload()
	local ret1 = uci_st:get("tm", "tm_statistics", "thisweek_upload")
	local ret2 = uci_st:get("tm", "tm_statistics", "thisweek_uploadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getThisweekDownload()
	local ret1 = uci_st:get("tm", "tm_statistics", "thisweek_download")
	local ret2 = uci_st:get("tm", "tm_statistics", "thisweek_downloadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getThisweekTotal()
	local ret1 = uci_st:get("tm", "tm_statistics", "thisweek_total")
	local ret2 = uci_st:get("tm", "tm_statistics", "thisweek_totalavg")
	local ret = ret1.."/"..ret2
	return ret
end
-- Thisweek tm_statistics End--

-- Thismonth tm_statistics Start--
function M.getThismonthConnectTime()
	local ret = uci_st:get("tm", "tm_statistics", "thismonth_conntime")
	return ret
end

function M.getThismonthUpload()
	local ret1 = uci_st:get("tm", "tm_statistics", "thismonth_upload")
	local ret2 = uci_st:get("tm", "tm_statistics", "thismonth_uploadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getThismonthDownload()
	local ret1 = uci_st:get("tm", "tm_statistics", "thismonth_download")
	local ret2 = uci_st:get("tm", "tm_statistics", "thismonth_downloadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getThismonthTotal()
	local ret1 = uci_st:get("tm", "tm_statistics", "thismonth_total")
	local ret2 = uci_st:get("tm", "tm_statistics", "thismonth_totalavg")
	local ret = ret1.."/"..ret2
	return ret
end
-- Thismonth tm_statistics End--

-- Lastmonth tm_statistics Start--
function M.getLastmonthConnectTime()
	local ret = uci_st:get("tm", "tm_statistics", "lastmonth_conntime")
	return ret
end

function M.getLastmonthUpload()
	local ret1 = uci_st:get("tm", "tm_statistics", "lastmonth_upload")
	local ret2 = uci_st:get("tm", "tm_statistics", "lastmonth_uploadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getLastmonthDownload()
	local ret1 = uci_st:get("tm", "tm_statistics", "lastmonth_download")
	local ret2 = uci_st:get("tm", "tm_statistics", "lastmonth_downloadavg")
	local ret = ret1.."/"..ret2
	return ret
end

function M.getLastmonthTotal()
	local ret1 = uci_st:get("tm", "tm_statistics", "lastmonth_total")
	local ret2 = uci_st:get("tm", "tm_statistics", "lastmonth_totalavg")
	local ret = ret1.."/"..ret2
	return ret
end
-- Lastmonth tm_statistics End--

function M.getWanMode()
	local ret = uci:get("network", "inet_global", "wan_mode")
        return ret
end

-- RAX30 www/tm_block/php/db/trafficMeter_limited_data.php
function M.getSpanVal_monthlyLimit()
	local volume_contrl = uci:get("tm", "tm_control", "volcontrl")
	local volume_monthly_limit = uci:get("tm", "tm_control", "volume_monthly_limit")
	local conntime_monthly_limit = uci:get("tm", "tm_control", "conntime_monthly_limit")

	if (volume_contrl == "true" ) then
	  return volume_monthly_limit
	else
	  return conntime_monthly_limit
	end
end

function M.getMlangVal_limitUnit()
	local volume_contrl = uci:get("tm", "tm_control", "volcontrl")

	if (volume_contrl == "true" ) then
	  ret = "volControl"
	else
	  ret = "timeControl"
	end

        if (ret == "volControl") then
	  return "ATM044"  --Mbytes
	elseif (ret == "timeControl") then
	  return "ATM043"  --hours
	end
end

function M.getTextVal_limitReachStatus()
	local ret = uci:get("tm", "tm_control", "warterMarkStatus")

	if (ret == "1" ) then
	  return "approach_limit"
	elseif (ret == "3" or ret == "4") then
	  return "reach_limit"
	else
	  return "not_reach_limit"
	end
end

return M

File: netgear/RAXE300/webs/tm_block/php/trafficMeter.php
<?
function getCheckboxVal_enable()
{   
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.traffic_on");
}

//tm = traffic meter
function getRadioVal_tm_type()
{
    $volume_contrl = db_get("Device.X_PEGATTRON_COM_TrafficMeter.volcontrl");
    $time_control = db_get("Device.X_PEGATTRON_COM_TrafficMeter.timecontrl");
    
    if ($volume_contrl == "true" && $time_control == "false") {
        return "volControl";
    }
    else if ($volume_contrl == "false" && $time_control == "true") {
        return "timeControl";
    }
}

function getSelectVal_volControlType()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.contrl_type");
    //"No limit"/"Download only"/"Both directions"
    
    //$contrl_type = db_get("Device.X_PEGATTRON_COM_TrafficMeter.contrl_type");
    
    //if ($contrl_type == "No limit") {
    //    return "1";
    //}
    //else if ($contrl_type == "Download only") {
    //    return "2";
    //}
    //else if ($contrl_type == "Both directions") {
    //    return "3";
    //}
}

function getTextVal_volMonthlyLimit()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.volume_monthly_limit");
}

function getTextVal_volRoundUp()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.round_up_volume");
}

function getTextVal_timeControlLimit()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.conntime_monthly_limit");
}

function getTextVal_counterHour()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.hour");
    //return "01";
}

function getTextVal_counterMin()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.min");
    //return "00";
}

function getSelectVal_amPmSel()
{
    // true:PM false:AM
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.ampm_sel");
}

function getSelectVal_counterDay()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.day");
}

function getTextVal_popMessage()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.waterMark");
}

function getCheckboxVal_turnLed()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.led_on");
}

function getCheckboxVal_disableInternet()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.block_on");
}

function getSpanVal_start_time()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.start_data_time");
}

function getSpanVal_current_time()
{
    //return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.current_data_time");
    //db value is not update instantly, so use date command directly
    return shell_exec("date +%c");
}

function getSpanVal_left_traffic()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.volume_left");
}

function getSpanVal_today_connect_time()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.today_conntime");
}

function getSpanVal_today_upload()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.today_upload");
}

function getSpanVal_today_download()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.today_download");
}

function getSpanVal_today_total()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.today_total");
}

function getSpanVal_yesterday_connect_time()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.yesterday_conntime");
}

function getSpanVal_yesterday_upload()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.yesterday_upload");
}

function getSpanVal_yesterday_download()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.yesterday_download");
}

function getSpanVal_yesterday_total()
{
	return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.yesterday_total");
}


function getSpanVal_thisweek_connect_time()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_conntime");
}

function getSpanVal_thisweek_upload()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_upload").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_uploadavg");
}

function getSpanVal_thisweek_download()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_download").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_downloadavg");
}

function getSpanVal_thisweek_total()
{
    return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_total").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thisweek_totalavg");
}



function getSpanVal_thismonth_connect_time()
{
	return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_conntime");
}

function getSpanVal_thismonth_upload()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_upload").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_uploadavg");
}

function getSpanVal_thismonth_download()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_download").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_downloadavg");
}

function getSpanVal_thismonth_total()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_total").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.thismonth_totalavg");
}


function getSpanVal_lastmonth_connect_time()
{
	return db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_conntime");
}

function getSpanVal_lastmonth_upload()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_upload").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_uploadavg");
}

function getSpanVal_lastmonth_download()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_download").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_downloadavg");
}

function getSpanVal_lastmonth_total()
{
	return  db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_total").'/'.db_get("Device.X_PEGATTRON_COM_TrafficMeter.Statistics.lastmonth_totalavg");
}

function get_wanMode()
{
    return db_get("Device.IP.X_PEGATRON_COM_IPv4Mode");
}

function getSpanVal_monthlyLimit()
{
    $tm_type = getRadioVal_tm_type();
    
    if ($tm_type == "volControl") {
        return db_get("Device.X_PEGATTRON_COM_TrafficMeter.volume_monthly_limit");
    }
    else if ($tm_type = "timeControl") {
        return db_get("Device.X_PEGATTRON_COM_TrafficMeter.conntime_monthly_limit");
    }
}

function getMlangVal_limitUnit()
{
    $tm_type = getRadioVal_tm_type();
    
    if ($tm_type == "volControl") {
        return "ATM044"; //Mbytes
    }
    else if ($tm_type = "timeControl") {
        return "ATM043";  //hours
    }
}

function getTextVal_limitReachStatus()
{
    $status = db_get("Device.X_PEGATTRON_COM_TrafficMeter.warterMarkStatus");
    
    if ($status == "1") {
        return "approach_limit";
    }
    else if($status == "3" || $status == "4") {
        return "reach_limit";
    }
    else {
        return "not_reach_limit";
    }
}
?>
